text/0000-target-spec-toml.md

```diff
@@ -0,0 +1,992 @@
+- **`crt-objects-fallback` -> N/A**
+  - Backwards compatible variant of `link-self-contained`
+  - Will be replaced by `linker.link_self_contained`
+  - Used in 306/306 built-in targets
+- **`data-layout` -> `backends.llvm.data_layout`**
+  - [Data layout][llvm-datalayout] to pass to LLVM
+  - Mandatory (if LLVM backend used), overrides
+  - Used in 306/306 built-in targets
+- **`linker-flavor` -> `linker.flavor`**
+  - Default linker flavour used if `-C linker-flavor` or `-C linker` are not
+    passed on the command line
+  - Optional, default depends on target-specific heuristics, overrides
+  - Used in 306/306 built-in targets
+- **`llvm-target` -> `backends.llvm.target`**
+  - Unversioned target tuple passed to LLVM
+  - Mandatory  (if LLVM backend used), overrides
+  - Used in 306/306 built-in targets
+- **`metadata.description` -> `metadata.description`**
+  - Human-readable description of the target, used in documentation generation
+  - Mandatory, does not inherit
+  - Used in 306/306 built-in targets
+- **`metadata.host_tools` -> `metadata.host_tools`**
+  - Whether this target is distributed with host tools
+  - Optional, defaults to `false`, does not inherit
+  - Used in 306/306 built-in targets
+- **`metadata.tier` -> `metadata.tier`**
+  - Tier of the target, used in documentation generation
+  - Mandatory, does not inherit
+  - Used in 306/306 built-in targets
+- **`metadata.std` -> N/A**
+  - Target's support for the standard library
+  - Will be replaced with `metadata.standard_library_support` in
+    [rfcs#3874][rfcs#3784-std-support]
+  - Used in 306/306 built-in targets
+- **`target-pointer-width` -> `target.pointer_width`**
+  - Number of bits in a pointer
+    - Influences `cfg(target_pointer_width)`
+  - Mandatory, overrides
+  - Used in 306/306 built-in targets
+- **`max-atomic-width` -> `target.maximum_atomic_width`**
+  - Maximum integer size in bits that this target can perform atomic operations on
+  - Optional, defaults to unset, overrides
+  - Used in 298/306 built-in targets
+- **`os` -> `os`**
+  - Operating system name to use for `cfg(target_os)`
+    - Combined with `architecture`, `env` and `vendor` to compute target tuple
+  - Mandatory, does not inherit
+  - Used in 255/306 built-in targets
+- **`target-family` -> `cfgs.family`**
+  - Values for `cfg(target_family)`
+    - Common options are `"unix"` and `"windows"`
+  - Optional, defaults to unset, overrides
+  - Used in 227/306 built-in targets
+- **`dynamic-linking` -> `linker.supports_dynamic`**
+  - Whether dynamic linking is available on this target
+  - Optional, defaults to `false`, overrides
+  - Used in 207/306 built-in targets
+- **`features` -> `backends.llvm.target_features`**
+  - Default target features to pass to LLVM
+    - These features overwrite `-Ctarget-cpu` but can be overwritten with
+      `-Ctarget-features`
+    - Corresponds to `llc -mattr=$features`
+    - Accepts LLVM feature names, not Rust feature names
```

Comment:
LLVM feature names do sometimes change, so I think we did have to do the same translation as we do for `-Ctarget-feature` and `#[target_feature(enable = "...")]`.